external-crates/move/crates/move-package-alt/src/package/root_package.rs

```diff
@@ -635,6 +633,48 @@ pkg_b = { local = "../pkg_b" }"#,
         assert_eq!(root.name(), &PackageID::from("graph"));
     }
 
+    #[test(tokio::test)]
+    async fn test_overriding_with_same_chain_id_is_ok() {
+        let project = test_utils::project()
+            .file(
+                "Move.toml",
+                &basic_manifest_with_env("graph", "0.0.1", DEFAULT_ENV_NAME, DEFAULT_ENV_ID),
+            )
+            .build();
+
+        let environment =
```

Comment:
should this fail even if we are loading for a different environment? I think as the code's written it should, but perhaps the test should check that case.